src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurStandardAccountExtractPdpEntryServiceImpl.java

```diff
@@ -196,21 +200,89 @@ private void addDetailEntriesToNewGroupEntry(PdpFeedGroupEntry newGroupEntry, Pd
+            LOG.info("cleanAccountingEntriesInDetailEntry, credits are greater than or equal to debits, so removing PDP transactions");
+            newDetailEntry.getAccounting().clear();
+        } else if (shouldDecductCreditsFromDebits(originalDebitTotal, originalCreditTotal)) {
+            LOG.info("cleanAccountingEntriesInDetailEntry, deduct credits from debits required");
+            deductCreditAmountsFromDebitTransactions(newDetailEntry, originalCreditTotal);
         } else {
-            LOG.info("addOnlyPositiveAccountingEntryForPDPProcessing, not adding accounting entry: " + originalAccountingEntry.toString());
+            LOG.info("cleanAccountingEntriesInDetailEntry, no changes required to the accounting entries");
+        }
+        removeNonPositiveAccountingEntriesFromDetailEntry(newDetailEntry);
+    }
+    
+    protected Map<DebitCreditTotal, KualiDecimal> caluclateTotals(PdpFeedDetailEntry detailEntry) {
+        KualiDecimal debitTotal = KualiDecimal.ZERO;
+        KualiDecimal creditTotal = KualiDecimal.ZERO;
+        for (PdpFeedAccountingEntry accountingEntry : detailEntry.getAccounting()) {
+            if (accountingEntry.getAmount().isPositive()) {
+                debitTotal = debitTotal.add(accountingEntry.getAmount());
+            } else {
+                creditTotal = creditTotal.add(accountingEntry.getAmount());
+            }
+        }
+        Map<DebitCreditTotal, KualiDecimal> totals = new HashMap<DebitCreditTotal, KualiDecimal>();
+        totals.put(DebitCreditTotal.CREDIT, creditTotal);
+        totals.put(DebitCreditTotal.DEBIT, debitTotal);
+        LOG.info("caluclateTotals, debitTotal: " + debitTotal + "  creditTotal: " + creditTotal) ;
+        return totals;
+    }
+    
+    protected enum DebitCreditTotal {
+        DEBIT,
+        CREDIT;
+    }
+    
+    private boolean shouldDecductCreditsFromDebits(KualiDecimal originalDebitTotal, KualiDecimal originalCreditTotal) {
+        return originalDebitTotal.isPositive() && originalCreditTotal.isNegative();
+    }
+
+    private boolean areThereMoreCreditsThanDebits(KualiDecimal originalDebitTotal, KualiDecimal originalCreditTotal) {
+        return originalCreditTotal.abs().isGreaterEqual(originalDebitTotal);
+    }
+
+    private void deductCreditAmountsFromDebitTransactions(PdpFeedDetailEntry newDetailEntry, KualiDecimal originalCreditTotal) {
+        KualiDecimal totalDeductionsLeft = new KualiDecimal(originalCreditTotal.doubleValue());
+        for (PdpFeedAccountingEntry accountingEntry : newDetailEntry.getAccounting()) {
+            if (accountingEntry.getAmount().isPositive() && totalDeductionsLeft.isNegative()) {
+                KualiDecimal newTransactionAmount = accountingEntry.getAmount().add(totalDeductionsLeft);
+                if (newTransactionAmount.isNegative()) {
+                    totalDeductionsLeft = new KualiDecimal(newTransactionAmount.doubleValue());
```

Comment:
Another spot where it would be better to directly assign the KualiDecimal reference, or to at least use the original object's BigDecimal value to create the new KualiDecimal.